### DISCUSSION TITLE
Remove console methods in production by using terser-webpack-plugin

### DISCUSSION THREAD

**Discussion by erlin1989** [vercel/next.js#12433] [Help] [answered]
I'm trying drop some of console methods in production by using terser-webpack-plugin with  options but getting some errors. Is there a way to pass (adding) options to the terser-webpack-plugin that nextjs already have.

**Comment by github0013** [comment]
```
yarn add -D babel-plugin-transform-remove-console
```

then

.babelrc
```
{
  "presets": ["next/babel"],
  "plugins": ["transform-remove-console"]
}
```

will remove `console.*`, but this doesn't limit only to production.

**Comment by github0013** [comment, marked as answer]
Oh, I think I got it.  If you write your `.babelrc` like this, it removes console.* only on build.

[env option][1]

```
{
  "presets": ["next/babel"],

  "env": {
    "production": {
      "plugins": ["transform-remove-console"]
    }
  }
}
```

[1]: https://babeljs.io/docs/en/6.26.3/babelrc#env-option

**Reply by erlin1989** [reply]
@github0013  I did it in that way thanks for your help!

**Reply by marklundin** [reply]
Is there a way to make this work with dynamic imports?

**Reply by fathomas** [reply]
@marklundin Did you find a solution for removing console.* from dynamically imported modules?

**Reply by aphofstede** [reply]
I think this will also remove console.log()s and console.error()s from your serverside code?

**Reply by github0013** [reply]
if you just want to stub only `console.log()` and only on the client side, running this code

```js
console.log = () => {}
```
somewhere at the top level of your client script might help?

**Reply by aphofstede** [reply]
I don't think the main goal there is to not _see_ console logs. Think bundle size, reverse engineering the code, obscure browser support, etc
